driver-core/src/test/functional/com/mongodb/client/CommandMonitoringTestHelper.java

```diff
@@ -133,6 +134,13 @@ public static void assertEventsEquality(final List<CommandEvent> expectedEvents,
 
                 assertEquals(expectedCommandStartedEvent.getDatabaseName(), actualCommandStartedEvent.getDatabaseName());
                 assertEquals(expectedCommandStartedEvent.getCommand(), actualCommandStartedEvent.getCommand());
+                if (((CommandStartedEvent) expected).getCommand().containsKey("recoveryToken")) {
+                    if (((CommandStartedEvent) expected).getCommand().get("recoveryToken").isNull()) {
+                        assertFalse(((CommandStartedEvent) actual).getCommand().containsKey("recoveryToken"));
+                    } else {
+                        assertTrue(((CommandStartedEvent) actual).getCommand().containsKey("recoveryToken"));
```

Comment:
How will this assertion ever succeed since in massageActualCommandSucceededEvent the recoverToken is unconditionally removed?

Reply:
Because `massageActualCommandSucceededEvent` returns a clone of the CommandEvent, and I am using the original in the check.